internal/templates/runasnonroot/template.go

```diff
@@ -0,0 +1,51 @@
+package runasnonroot
+
+import (
+	"fmt"
+
+	"golang.stackrox.io/kube-linter/internal/check"
+	"golang.stackrox.io/kube-linter/internal/diagnostic"
+	"golang.stackrox.io/kube-linter/internal/extract"
+	"golang.stackrox.io/kube-linter/internal/lintcontext"
+	"golang.stackrox.io/kube-linter/internal/objectkinds"
+	"golang.stackrox.io/kube-linter/internal/templates"
+)
+
+func init() {
+	templates.Register(check.Template{
+		Name:        "run-as-non-root",
+		Description: "Flag containers without runAsUser specified",
+		SupportedObjectKinds: check.ObjectKindsDesc{
+			ObjectKinds: []string{objectkinds.DeploymentLike},
+		},
+		Parameters: nil,
+		Instantiate: func(_ map[string]string) (check.Func, error) {
+			return func(_ *lintcontext.LintContext, object lintcontext.Object) []diagnostic.Diagnostic {
+				podSpec, found := extract.PodSpec(object.K8sObject)
+				if !found {
+					return nil
+				}
+				var results []diagnostic.Diagnostic
+				var podSCValue bool
+				if podSpec.SecurityContext != nil && podSpec.SecurityContext.RunAsNonRoot != nil {
```

Comment:
I wonder if it wouldn't be simpler to work with `unstructured` objects in order to allow nested and uniform (between pod/container) element access...

Reply:
Yeah, I've been thinking about that too. Still deciding whether to refactor it. The type-safety is definitely nice, but it doesn't feel _super_ helpful given that you need to be so careful with the `nil` dereferences.